api/app/graphql/mutations/update_task.rb

```diff
@@ -5,9 +5,15 @@ class UpdateTask < BaseMutation
     argument :task_id, ID, required: true, loads: Types::TaskType
     argument :name, String, required: true
     argument :complete, Boolean, required: true
+    argument :user_id, ID, required: true
+    argument :duedate, GraphQL::Types::ISO8601Date, required: false
 
-    def resolve(task:, name:, complete:)
-      task.update!(name: name, complete: complete)
+    def resolve(task:, name:, complete:, user_id:, duedate:)
+      if task.complete != complete && complete == true
```

Comment:
At the very least you can rewrite this line as `if task.complete != complete && complete`
The `== true` part is redundant.

There is some Rails magic we could use to make this method a bit cleaner. We can touch on it tomorrow.